bundle/src/test/java/com/adobe/acs/commons/properties/filter/TemplatedFilterTest.java

```diff
@@ -0,0 +1,272 @@
+ * %%
+ * Copyright (C) 2020 Adobe
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.adobe.acs.commons.properties.filter;
+
+import io.wcm.testing.mock.aem.junit.AemContext;
+import org.apache.sling.testing.mock.sling.ResourceResolverType;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.adobe.acs.commons.properties.TemplatedDialogTestUtil.defaultService;
+import static org.hamcrest.Matchers.startsWith;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.mock;
+
+@RunWith(MockitoJUnitRunner.class)
+public class TemplatedFilterTest {
+    @Rule
+    public final AemContext context = new AemContext(ResourceResolverType.JCR_OAK);
+
+    @Rule
+    public ExpectedException thrown = ExpectedException.none();
+
+    @Mock
+    private FilterChain filterChain;
+
+    private TemplatedFilter filter;
+
+    @Before
+    public void setup() throws IOException, ServletException {
+        context.load().json(getClass().getResourceAsStream("TemplatedFilterContent.json"), "/content/we-retail/language-masters/en/experience");
+        context.currentResource("/content/we-retail/language-masters/en/experience/arctic-surfing-in-lofoten/jcr:content/root/hero_image");
+    }
+
+    @Test
+    public void testNonJsonResponse() throws IOException, ServletException {
+        initServices(null);
+        context.requestPathInfo().setExtension("html");
+
+        String before = "<html><body>{{page_properties.jcr:title}}</body></html>";
+        doAnswer(invocation -> {
+            HttpServletResponse response =
+                    (HttpServletResponse) invocation.getArguments()[1];
+            response.getWriter().println(before);
+            return null;
+        }).when(filterChain).doFilter(any(), any());
+
+        filter.doFilter(context.request(), context.response(), filterChain);
+        assertTrue(startsWith(before)
```

Comment:
Out of curiosity, does a simple `assertEquals` fail here?

Reply:
It does actually - it automatically adds a newline character at the end (for some reason)